code/core/src/manager/components/TourGuide/TourGuide.tsx

```diff
@@ -0,0 +1,200 @@
+          </>
+        ),
+        onNext: step.onNext ? () => step.onNext?.({ next }) : onNext && (() => onNext?.({ next })),
+      };
+    });
+  }, [steps, onNext, updateStepIndex]);
+
+  const callback = useCallback(
+    (data: CallBackProps) => {
+      if (data.action === ACTIONS.NEXT && data.lifecycle === 'complete') {
+        if (data.index === data.size - 1) {
+          onComplete?.();
+        } else if (data.step?.onNext) {
+          data.step.onNext();
+        } else {
+          updateStepIndex(data.index + 1);
+        }
+      }
+      if (data.action === ACTIONS.CLOSE) {
+        onDismiss?.();
+      }
+    },
+    [onComplete, onDismiss, updateStepIndex]
+  );
+
+  if (stepIndex === null) {
+    return null;
+  }
+
+  return (
+    <Joyride
+      continuous
+      steps={mappedSteps}
+      stepIndex={stepIndex}
+      spotlightPadding={0}
+      disableCloseOnEsc
+      disableOverlayClose
+      disableScrolling
+      callback={callback}
+      tooltipComponent={TourTooltip}
+      floaterProps={{
+        disableAnimation: true,
+        styles: {
+          arrow: {
+            length: 20,
+            spread: 2,
+          },
+          floater: {
+            filter:
+              theme.base === 'light'
+                ? 'drop-shadow(0px 5px 5px rgba(0,0,0,0.05)) drop-shadow(0 1px 3px rgba(0,0,0,0.1))'
+                : 'drop-shadow(#fff5 0px 0px 0.5px) drop-shadow(#fff5 0px 0px 0.5px)',
+          },
+        },
+      }}
+      styles={{
+        overlay: {
+          mixBlendMode: 'unset',
+          backgroundColor: steps[stepIndex]?.target === 'body' ? 'rgba(27, 28, 29, 0.2)' : 'none',
+        },
+        spotlight: {
+          backgroundColor: 'none',
+          border: `solid 2px ${theme.base === 'light' ? theme.color.secondary : darken(0.18, theme.color.secondary)}`,
+          boxShadow: '0px 0px 0px 9999px rgba(27, 28, 29, 0.2)',
+        },
+        tooltip: {
+          width: 280,
+          color: theme.color.lightest,
+          background:
+            theme.base === 'light' ? theme.color.secondary : darken(0.18, theme.color.secondary),
+        },
+        options: {
+          zIndex: 9998,
+          primaryColor:
+            theme.base === 'light' ? theme.color.secondary : darken(0.18, theme.color.secondary),
+          arrowColor:
+            theme.base === 'light' ? theme.color.secondary : darken(0.18, theme.color.secondary),
+        },
+      }}
+    />
+  );
+};
+
+let root: ReturnType<typeof createRoot> | null = null;
+
+TourGuide.render = (props: ComponentProps<typeof TourGuide>) => {
+  let container = document.getElementById('storybook-tour');
+  if (!container) {
+    container = document.createElement('div');
+    container.id = 'storybook-tour';
+    document.body.appendChild(container);
+  }
+  root = root ?? createRoot(container);
+  root.render(
+    <ThemeProvider theme={convert(themes.light)}>
```

Comment:
A hard-coded `themes.light` surely is not correct?

Reply:
Guided tours don't have dark mode. Or light mode for that matter. It's just white text on blue tooltips.